cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,201 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieSerializer, MovieListSerializer, MovieDetailSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+def sample_movie(**params) -> Movie:
+    defaults = {
+        "title": "The Great Brick",
+        "description": "The Great Brick fall on superman head",
+        "duration": 121
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+class TestUnauthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
```

Comment:
This is a bug. You're testing that an unauthenticated user receives a 401 status code when trying to access the movie list. However, in the `setUp` method, you didn't set the client to be unauthenticated. You need to call `self.client.force_authenticate(user=None)` to ensure the client is unauthenticated.